vespa/src/main/java/ai/marqo/search/HybridSearcher.java

```
@@ -459,4 +545,71 @@ static String extractDocIdFromHitId(String fullPath) {
+                            + " hit.",
+                    verbose);
+            return hits;
+        }
+        hitMatchFeatures = (FeatureData) hits.get(0).getField("matchfeatures");
+        if (hitMatchFeatures.getDouble("global_mult_modifier") == null
+                || hitMatchFeatures.getDouble("global_add_modifier") == null) {
+            logIfVerbose(
+                    "SKIPPING applying global score modifiers. First hit is missing either "
+                            + "global_mult_modifier or global_add_modifier match-feature.",
+                    verbose);
+            return hits;
+        }
+
+        for (Hit hit : hits) {
+            logIfVerbose("Applying score modifiers to hit: " + hit.getId(), verbose);
+            // Extract the mult and add modifiers from match-features
+            hitMatchFeatures = (FeatureData) hit.getField("matchfeatures");
+            if (hitMatchFeatures != null) {
+                mult_modifier = hitMatchFeatures.getDouble("global_mult_modifier");
+                add_modifier = hitMatchFeatures.getDouble("global_add_modifier");
+
+                if (mult_modifier != null && add_modifier != null) {
+                    // Apply the modifiers to the hit's relevance
+                    original_score = hit.getRelevance().getScore();
+                    modified_score = original_score * mult_modifier + add_modifier;
+                    logIfVerbose(
+                            String.format(
+                                    "Original score: %.7f, mult modifier: %.2f, add modifier: %.2f,"
+                                            + " Modified score: %.7f",
+                                    original_score, mult_modifier, add_modifier, modified_score),
+                            verbose);
+                    hit.setRelevance(modified_score);
+                } else {
+                    throw new RuntimeException(
+                            "Failed to apply global score modifiers. Hit "
```

nit. using fail fast pattern can save us from too deeply nested code structure.

---

Answered this in another comment. We fully skip the score modifiers step if `marqo__mult_weights_global` and `marqo__add_weights_global` are not set or empty. This extra check is just to fail with a good error if one of the hits is missing a required field for global score modifiers